controllers/perconaservermysqlbackup_controller.go

```diff
@@ -308,30 +308,31 @@ func getDestination(storage *apiv1alpha1.BackupStorageSpec, clusterName, creatio
 	return dest
 }
 
-func (r *PerconaServerMySQLBackupReconciler) getBackupSources(ctx context.Context, cluster *apiv1alpha1.PerconaServerMySQL) ([]string, error) {
-	l := log.FromContext(ctx).WithName("getBackupSources")
+func (r *PerconaServerMySQLBackupReconciler) getBackupSource(ctx context.Context, cluster *apiv1alpha1.PerconaServerMySQL) (string, error) {
+	l := log.FromContext(ctx).WithName("getBackupSource")
 
 	operatorPass, err := k8s.UserPassword(ctx, r.Client, cluster, apiv1alpha1.UserOperator)
 	if err != nil {
-		return nil, errors.Wrap(err, "get operator password")
+		return "", errors.Wrap(err, "get operator password")
 	}
 
 	top, err := topology.Get(ctx, cluster, operatorPass)
 	if err != nil {
-		return nil, errors.Wrap(err, "get topology")
+		return "", errors.Wrap(err, "get topology")
 	}
 
-	var sources []string
+	var source string
 	if len(top.Replicas) < 1 {
-		sources = append(sources, top.Primary)
+		source = top.Primary
 		l.Info("no replicas found, using primary as the backup source", "primary", top.Primary)
 	} else {
 		for _, replica := range top.Replicas {
-			sources = append(sources, replica)
+			source = replica
+			break
 		}
```

Comment:
why don't we just assign `source = top.Replicas[0]` which is effectively the same thing

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/215/commits/11d2d9aad16b04ff25c6d2d9597014cc36236af3